recipes-core/systemd/systemd/229/0002-nspawn-added-support-for-starting-DHCP-client-in-non.patch

```diff
@@ -0,0 +1,203 @@
+@@ -538,3 +540,87 @@ int veth_extra_parse(char ***l, const char *p) {
+         a = b = NULL;
+         return 0;
+ }
++
++int spawn_dhcp_client(const char *dhcp_client, pid_t *rpid) {
++        pid_t pid;
++        const char *interface = "host0";
++        const char *argv[16];
++
++        pid = fork();
++        if (pid < 0)
++                return log_error_errno(errno, "Failed to fork dhclient: %m");
++        else if (pid == 0) {
++                int nullfd;
++                char *empty_env = NULL;
++
++                nullfd = open("/dev/null", O_RDWR);
++                if (nullfd < 0)
++                        _exit(EXIT_FAILURE);
++
++                if (dup3(nullfd, STDIN_FILENO, 0) < 0)
++                        _exit(EXIT_FAILURE);
++
++                if (dup3(nullfd, STDOUT_FILENO, 0) < 0)
++                        _exit(EXIT_FAILURE);
++
++                if (dup3(nullfd, STDERR_FILENO, 0) < 0)
++                        _exit(EXIT_FAILURE);
++
++                if (nullfd > 2)
++                        safe_close(nullfd);
++
++                (void) reset_all_signal_handlers();
++                (void) reset_signal_mask();
++                close_all_fds(NULL, 0);
++
++                if (dhcp_client) {
++                        char **args = strv_split(dhcp_client, " ");
++                        char **a;
++                        int i;
++
++                        if (args == NULL)
++                                _exit(EXIT_FAILURE);
++
++                        i = 0;
++                        STRV_FOREACH(a, args) {
++                                if (i >= (int)ELEMENTSOF(argv) - 1)
++                                        _exit(EXIT_FAILURE);
++
++                                if (i == 0) {
++                                        char *p = strrchr(*a, '/');
++
++                                        argv[i] = p ? p + 1 : *a;
++                                }
++                                else
++                                        argv[i] = *a;
++                        }
++                        argv[i] = interface;
++
++                        execve(args[0], (char *const *)argv, &empty_env);
++                        _exit(EXIT_FAILURE);
++                }
++                else {
++                        argv[0] = "dhclient";
++                        argv[1] = interface;
++                        argv[2] = NULL;
++
++                        execve("/usr/sbin/dhclient", (char *const*)argv, &empty_env);
++                        execve("/sbin/dhclient", (char *const*)argv, &empty_env);
++
++                        argv[0] = "udhcpc";
++                        argv[1] = "-i";
++                        argv[2] = interface;
++                        argv[3] = NULL;
++
++                        execve("/usr/sbin/udhcpc", (char *const*)argv, &empty_env);
++                        execve("/sbin/udhcpc", (char *const*)argv, &empty_env);
```

Comment:
it would be worth to see what's upstream approach to this. In my mind we should create an automatic service based on `systemd-networkd` inside the container. One problem would be to make sure that binary is available inside the container, but we could bind-mount it. And as I said, making it a service, rather than manually spawning it.